shell/browser/api/electron_api_media_session.h

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2022 Slack Technologies, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ELECTRON_SHELL_BROWSER_API_ELECTRON_API_MEDIA_SESSION_H_
+#define ELECTRON_SHELL_BROWSER_API_ELECTRON_API_MEDIA_SESSION_H_
+
+#include <vector>
+
+#include "base/callback.h"
+#include "base/values.h"
+#include "gin/arguments.h"
+#include "gin/handle.h"
+#include "gin/wrappable.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "services/media_session/public/mojom/media_session.mojom.h"
+#include "shell/browser/event_emitter_mixin.h"
+
+namespace content {
+class MediaSession;
+}  // namespace content
+
+namespace electron {
+
+namespace api {
+
+class MediaSession : public gin::Wrappable<MediaSession>,
+                     public gin_helper::EventEmitterMixin<MediaSession>,
+                     public media_session::mojom::MediaSessionObserver {
+ public:
+  static gin::Handle<MediaSession> Create(v8::Isolate* isolate,
+                                          content::MediaSession* media_session);
+
+  // gin::Wrappable
+  static gin::WrapperInfo kWrapperInfo;
+  gin::ObjectTemplateBuilder GetObjectTemplateBuilder(
+      v8::Isolate* isolate) override;
+  const char* GetTypeName() override;
+
+  // disable copy
+  MediaSession(const MediaSession&) = delete;
+  MediaSession& operator=(const MediaSession&) = delete;
+
+ protected:
+  MediaSession(v8::Isolate* isolate, content::MediaSession* media_session);
+  ~MediaSession() override;
```

Comment:
Why not make these private? Are we planning on creating subclasses of this later on?

Reply:
This file initially started by duplicating https://github.com/electron/electron/blob/main/shell/browser/api/electron_api_debugger.h

I'm not too sure whether it's necessary or not. I see the same thing in [gin::Wrappable](https://source.chromium.org/chromium/chromium/src/+/main:gin/wrappable.h;l=105-107;drc=68c0798d46c47105ffdbf40c58bb63153a148116).

@deepak1556 Do you happen to recall the reasoning for setting this to protected? I realize the `Debugger` API was introduced quite a while ago now. 😅

Reply:
Ah, I believe the Debugger class ctor can be turned private too. However, Wrappable is subclassed at multiple places, like https://github.com/electron/electron/blob/84cf685e761b2dd22f064e0dbf33b290bbec5fc0/shell/browser/api/electron_api_auto_updater.h#L20 for example, so this wouldn't be applicable for the Wrappable class.